### PR TITLE
Validation mixup

### PR DESCRIPTION
**Pull request by marco-bazzani**
- Adds mixup to validation
- Rounds mixup targets up (what interval do we want? current default is 0.5)

### CODE REVIEW

File: pyha_analyzer/augmentations.py [hunk 1]
```diff
@@ -8,12 +8,35 @@
...
+    return dict(zip(probabilities, seq))
+
+def sample(distribution: Dict[numbers.Integral, numbers.Integral]):
+    return np.random.choice(distribution.values(), p = distribution.keys())
+
+l = [1, 2, 3, 4, 5]
+dist = hyperbolic(l)
+print(f"{dist=}")
+for i in range(1000):
+    sample(dist)
```
Comment (benjamin-cates) [inline pyha_analyzer/augmentations.py]: slayyy I love committing test code

File: pyha_analyzer/test.py [hunk 2]
```diff
@@ -36,24 +36,26 @@ def test_augs(self):
...
+        num_clips_range = list(range(cfg.mixup_num_clips_range[0],
+                                     cfg.mixup_num_clips_range[1] + 1))
         mixup = Mixup(dataset.samples, dataset.class_to_idx, cfg)
         new_audio, new_label = mixup(audio, label)
         # Assert mixup output
         assert new_audio.shape == audio.shape, "Mixup should not change shape"
         assert new_label[label_id] == 1., \
                 "Mixup labels should be rounded up to nearest interval"
-        assert new_label.sum() == 2., "Mixup label should add to two 2"
+        assert float(new_label.sum()) in num_clips_range, "Mixup label should add to two 2"
```
Comment (benjamin-cates) [inline pyha_analyzer/test.py]: "add to two 2" :(

File: pyha_analyzer/augmentations.py [hunk 3]
```diff
@@ -96,10 +110,10 @@ def forward(
         chosen clip, Tensor of target mixed with the
         target of the randomly chosen file
         """
-        alpha = utils.rand(*self.alpha_range)
         if utils.rand(0,1) < self.prob:
             return clip, target
 
+        print(self.num_clips_distribution)
```
Comment (benjamin-cates) [inline pyha_analyzer/augmentations.py]: Print statements in forward UwU

File: pyha_analyzer/test.py (hunk 2 above)
Comment (benjamin-cates) [reply pyha_analyzer/test.py]: also inaccurate with num_clips_range now

File: pyha_analyzer/augmentations.py (hunk 1 above)
Comment (marco-bazzani) [reply pyha_analyzer/augmentations.py]: I had to transfer to a new machine

Comment (marco-bazzani) [reply pyha_analyzer/augmentations.py]: Leave me alone

**Comment by marco-bazzani** [comment]
The type issues aint in my code

**Comment by marco-bazzani** [comment]
TODO: Decide whether to do fancier clip mixing ratios

**Review by Sean1572 — DISMISSED** [review]
lgtm

**Comment by benjamin-cates** [comment]
image
This function isn't working for me

**Comment by benjamin-cates** [comment]
Your definition of sample does not work if two elements have the same probability. Also hyperbolic has the same problem if there are duplicate integers. Perhaps it would be much better to have Dict[int,float] instead of Dict[float,int]. Although you would have to manage adding duplicate keys instead of overwriting. Might have to be List[Tuple[float,int]]
image

File: pyha_analyzer/augmentations.py [hunk 4]
```diff
@@ -5,15 +5,55 @@
...
+        raise ValueError('Passed iterable cannot contain zero')
+    return [1/x for x in seq]
+
+def hyperbolic(seq: Iterable[int]) -> Dict[float, int]:
+    """
+    Takes a list of numbers and assigns them a probability
+    distribution accourding to the inverse of their values
+    """
+    norm_factor = sum(invert(seq))
+    probabilities = [x/norm_factor for x in invert(seq)]
```
Comment (benjamin-cates) [inline pyha_analyzer/augmentations.py]: Should save the output of invert instead of calling it twice :3

File: pyha_analyzer/augmentations.py [hunk 5]
```diff
@@ -5,15 +5,55 @@
...
+    """
+    Sample single value from distribution
+    distribution.keys() should be the probabilities,
+    and distribution.values() should be the values
+    """
+    probabilities = list(distribution.keys())
+    values = list(distribution.values())
+    return np.random.choice(values, p = probabilities)
+
+def gen_uniform_values(n: int):
```
Comment (benjamin-cates) [inline pyha_analyzer/augmentations.py]: I was gonna add a docstring here but I don't know what the goal of this function is lol

File: .pylintrc [hunk 6]
```diff
@@ -178,6 +178,7 @@ function-naming-style=snake_case
 good-names=i,
            j,
            k,
+	   n,
```
Comment (benjamin-cates) [inline .pylintrc]: Indentation 😈 

It doesn't matter just kinda funny

**Review by benjamin-cates — APPROVED** [review]
I'm happy with it, hopefully no merge conflicts 🤞

**Pull request merged**